### PR TITLE
Add helper function to convert channel str to channel object

### LINKED ISSUE #2876: Enhancement of channel specification in pulse visualization
### What is the expected enhancement?
We need to create `PulseChannelSpec` instance to specify `channels_to_plot` option of pulse visualizer. This is cumbersome to visualize a schedule created by scheduler, or a schedule automatically created by some python callables - in the case when `PulseChannelSpec` is internally configured. Using channel letters instead of `PulseChannel` will improve usability.

(now)
```python
schedule = create_some_schedule(*args, **kwargs)

from qiskit import pulse
system = pulse.PulseChannelSpec.from_backend(backend)
channels = [system.drives[0], system.measures[0]]

schedule.draw(channels_to_plot=channels)
```

(what I'm proposing)
```python
schedule = create_some_schedule(*args, **kwargs)

schedule.draw(channels_to_plot=['d0', 'm0'])
```

This might neglect type checking policy in pulse module.

### PR DESCRIPTION
**Pull request by nkanazawa1989**
### Summary
This PR adds helper function to convert list of channel letters to corresponding channel object. This improves usability of pulse visualization especially when we visualize automatically created schedule, e.g. CmdDef and scheduler.

close #2876

### CODE REVIEW

**Comment by taalexander** [comment]
I am still not sure if adding this feature is worth the added inconvenience of maintaining additional code in the codebase. It also ties us to a specific naming convention for channels moving forward. Thoughts @lcapelluto?

File: qiskit/visualization/pulse/utils.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    Args:
+        channels_str (list): List of channel letters.
+
+    Returns:
+        channel_to_plot: List of channel object instance corresponding to the input.
+
+    Raises:
+        VisualizationError: when invalid channel is specified.
+    """
+    defined_channels = [DriveChannel, MeasureChannel, ControlChannel, AcquireChannel]
```
[the remark this answers is not in the document]
Comment (nkanazawa1989) [reply qiskit/visualization/pulse/utils.py]: thanks. it looks much more simple.

**Comment by nkanazawa1989** [comment]
@lcapelluto thank you for reviewing! My point is, importing and creating channels objects every single time checking the schedule are very stressful work. using strings can enable this visualization by a single line, and pulse programming will become much more efficient - by default, all active channels are plotted in a single figure, and usually it is difficult to check the waveform of pulses I'm interested in.

I understand pros and cons of using these strings (and I realized these string labels are designed to be hidden from users, and should exist only in qobj). any good ideas for selecting channels to be visualized?

**Comment by nkanazawa1989** [comment]
This is good suggestion. `schedule.channels` may be useful, but currently it returns a tuple of channels and still it's not smart - we need to show its value and manually count the index of tuple on screen.

I think adding some feature to filter channels in `schedule.channels` is a good direction.

**Comment by nkanazawa1989** [comment]
Ok, that is the solution.
1. filter the schedule before calling draw method.
2. draw the schedule.

I'll close this PR. but the solution is bit tricky, so let me update pulse tutorial.

**Comment by taalexander** [comment]
> Ok, that is the solution.
> 
> 1. filter the schedule before calling draw method.
> 2. draw the schedule.
> 
> I'll close this PR. but the solution is bit tricky, so let me update pulse tutorial.

For backwards compatibility you could still keep `channels_to_plot` and call `filter` internally within the method.

**Pull request closed without merging**